ocs_ci/ocs/resources/mcg.py

```diff
@@ -272,3 +296,289 @@ def _retrieve_reduction_data():
                 'Not enough data reduction. Something is wrong.'
             )
             return False
+
+    def request_aws_credentials(self):
+        """
+        Uses a CredentialsRequest CR to create an AWS IAM that allows the program
+        to interact with S3
+
+        Returns:
+            OCS: The CredentialsRequest resource
+        """
+        awscreds_data = templating.load_yaml(constants.MCG_AWS_CREDS_YAML)
+        req_name = create_unique_resource_name('awscredreq', 'credentialsrequests')
+        awscreds_data['metadata']['name'] = req_name
+        awscreds_data['metadata']['namespace'] = self.namespace
+        awscreds_data['spec']['secretRef']['name'] = req_name
+        awscreds_data['spec']['secretRef']['namespace'] = self.namespace
+        creds_request = create_resource(**awscreds_data)
+
+        def _retrieve_credreq_provision_state():
+            return run_cmd(
+                f"oc get credentialsrequests {creds_request.name} -o jsonpath={{.status.provisioned}}"
+            )
+
+        try:
+            for provisioned_state in TimeoutSampler(30, 5, _retrieve_credreq_provision_state):
+                if provisioned_state == "true":
+                    sleep(10)
```

Comment:
Not sure how TimeoutSampler behaves if you're waiting here for 10 seconds while its polling rate is 5 seconds.

Reply:
If I reach provisioned_state, I sleep for 10 seconds and then break - meaning it has no effect over `TimeoutSampler`. This sleep is needed because there is absolutely no way to check whether the IAM was actually provisioned except trying to use it (which causes an exception). This is just a safety measure that allows it to truly finish setting up. After the sleep, no further samplers are used.

Reply:
ok. Thanks for the explanation. 
Do we expect more than 1 iteration of TimeoutSampler?

Reply:
Theoretically - it's possible.
Practically - I never ran into a case in which it happened.

Reply:
So maybe the TimeoutSampleris redundant here?

Reply:
I think it'd be better to go on the safer side of things - if it won't be used - then no harm is done. If it'll be used - removing it will cause an exception.

It's mostly here because I saw that the NooBaa sanity test relied upon it

Reply:
Took out that TimeoutSampelr - resolved :)